File: app/src/main/java/com/thecoolguy/rumaan/fileio/data/db/UploadHistoryRoomDatabase.java
package com.thecoolguy.rumaan.fileio.data.db;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import android.content.Context;
import com.thecoolguy.rumaan.fileio.data.models.FileEntity;

@Database(entities = { FileEntity.class }, version = 1)
public abstract class UploadHistoryRoomDatabase extends RoomDatabase {

  private static UploadHistoryRoomDatabase sINSTANCE;

  public static UploadHistoryRoomDatabase getInstance(Context context) {
    if (sINSTANCE == null) {
      synchronized (UploadHistoryRoomDatabase.class) {
        if (sINSTANCE == null) {
          sINSTANCE = Room.databaseBuilder(context.getApplicationContext(),
              UploadHistoryRoomDatabase.class,
              "upload_history")
              .build();
        }
      }
    }
    return sINSTANCE;
  }

  public abstract UploadItemDao uploadItemDao();
}

File: app/src/main/java/com/thecoolguy/rumaan/fileio/data/db/UploadHistoryRoomDatabase.kt
package com.thecoolguy.rumaan.fileio.data.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.thecoolguy.rumaan.fileio.data.models.FileEntity

@Database(entities = [FileEntity::class], version = 1)
abstract class UploadHistoryRoomDatabase : RoomDatabase() {

    abstract fun uploadItemDao(): UploadItemDao

    companion object {
        private lateinit var INSTANCE: UploadHistoryRoomDatabase

        fun getInstance(context: Context): UploadHistoryRoomDatabase {
            synchronized(UploadHistoryRoomDatabase::class.java) {
                INSTANCE = Room.databaseBuilder(context.applicationContext,
                        UploadHistoryRoomDatabase::class.java,
                        "upload_history")
                        .allowMainThreadQueries()
                        .build()
            }
            return INSTANCE
        }
    }
}
